pytest_asyncio/plugin.py

```diff
@@ -486,9 +496,13 @@ def pytest_runtest_setup(item: pytest.Item) -> None:
 @pytest.fixture
 def event_loop(request: "pytest.FixtureRequest") -> Iterator[asyncio.AbstractEventLoop]:
     """Create an instance of the default event loop for each test case."""
-    loop = asyncio.get_event_loop_policy().new_event_loop()
-    yield loop
-    loop.close()
+    yield asyncio.get_event_loop_policy().new_event_loop()
+    # Call the garbage collector to trigger ResourceWarning's as soon
+    # as possible (these are triggered in various __del__ methods).
+    # Without this, resources opened in one test can fail other tests
+    # when the warning is generated.
+    gc.collect()
```

Comment:
A bunch of people implement their own `event_loop` fixture, in order to control fixture scope. If we change the fixture definition, we'd have to inform them to adjust as well.

Asking users to remove `loop.close()` to benefit from the new cleanup code is an easy sell. Adding a call to `gc.collect` in the event_loop fixture not so much.

Is there away around the `gc.collect` call in the fixture code?

Reply:
I suppose the call to `gc.collect` isn't directly related to task cleanup. I added this in my codebase because there were some tests that opened sockets and never closed them. Having the garbage collector run during the test teardown would make sure that the correct test failed when running pytest with `-W error`. But this is unrelated to pytest-asyncio so maybe it doesn't belong here.